alpine/LoadBalancer.hpp

```diff
@@ -12,30 +12,34 @@ class LoadBalancer{
         double loadbalancethreshold_m;
         Field_t<Dim>* rho_m;
         VField_t<T, Dim>* E_m;
+        Field<T, Dim> *phi_m;
         std::shared_ptr<ParticleContainer<T, Dim>> pc_m;
         std::shared_ptr<FieldSolver_t> fs_m;
         unsigned int loadbalancefreq_m;
         ORB<T, Dim> orb;
     public:
         LoadBalancer(double lbs, std::shared_ptr<FieldContainer<T,Dim>> &fc, std::shared_ptr<ParticleContainer<T, Dim>> &pc, std::shared_ptr<FieldSolver_t> &fs)
-           :loadbalancethreshold_m(lbs), rho_m(&fc->getRho()), E_m(&fc->getE()), pc_m(pc), fs_m(fs) {}
+           :loadbalancethreshold_m(lbs), rho_m(&fc->getRho()), E_m(&fc->getE()), phi_m(&fc->getPhi()), pc_m(pc), fs_m(fs) {}
 
         ~LoadBalancer() {  }
```

Comment:
This entire class could also be part of a common ALPINE-level manager superclass. Depending on how the PIC manager is intended to work, it could also be moved all the way up to the generic PIC manager class and accessed via generic interface. Aside from the behavior specific to the uniform plasma test, the partitioning code itself is not application-specific.

Reply:
I agree. @srikrrish @matt-frey do you agree with putting LoadBalancer in the src/Manger/ directory?

Reply:
Yes if it is a generic enough code which can be used across applications it can be in src to avoid code repetition.

Reply:
Sri and I thought the load balance class to be a generic base class for all particle load balancing methods. It should therefore not be in Alpine. However, don't put this into src/Manager.

Reply:
@matt-frey OK. Where should I put it then?

Reply:
I thought we have a directory called `LoadBalancer` (or similar). If not, we may do this. We can also do this in further optimisations as @srikrrish already posted below.

Reply:
I don't see dir called "LoadBalancer" in src. We have [Decomposition](https://github.com/IPPL-framework/ippl/tree/pic-manager/src/Decomposition). I guess it should go there. I agree. Let's do it in another PR.